internal/controller/instaslice_controller.go

```diff
@@ -658,9 +658,25 @@ func (r *InstasliceReconciler) setupWithManager(mgr ctrl.Manager) error {
 	if err != nil {
 		return err
 	}
+	// pod filtering
+	// only watches Pods that: have the mutation label
+	// Use event-based predicate with label checks and update detection
+	instaslicePredicate := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return e.Object != nil && e.Object.GetLabels()[LabelInstasliceMutated] == InstaslicePodMutatedTrue
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return (e.ObjectNew != nil && e.ObjectNew.GetLabels()[LabelInstasliceMutated] == InstaslicePodMutatedTrue) ||
```

Comment:
Should this be 
- Object not nil and
- Pod mutated and
- ResourceVersion doesn't match
Something like
```suggestion
			return (e.ObjectNew != nil && e.ObjectNew.GetLabels()[LabelInstasliceMutated] == InstaslicePodMutatedTrue) &&
```